sdk/identity/identity/src/credentials/managedIdentityCredential/imdsMsi.ts

```diff
@@ -48,7 +48,7 @@ function prepareRequestOptions(resource?: string, clientId?: string): PipelineRe
   const query = qs.stringify(queryParameters);
 
   return {
-    url: process.env.AZURE_POD_IDENTITY_TOKEN_URL ?? `${imdsEndpoint}?${query}`,
+    url: `${process.env.AZURE_POD_IDENTITY_AUTHORITY_HOST ?? imdsHost}${imdsEndpointPath}?${query}`,
```

Comment:
If $AZURE_POD_IDENTITY_AUTHORITY_HOST is `https://foo/` you'll end up with `https://foo//metadata/identity/oauth2/token`

Reply:
good point! thanks!

Reply:
Fixed here: https://github.com/Azure/azure-sdk-for-js/pull/16682/commits/be78f8cd478791e53a37d8da2486b67912312490